src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -494,23 +494,38 @@ export function promptForBoolean( message: string, initial: boolean ): Promise<b
 	return Promise.resolve( initial );
 }
 
-function resolvePhpVersion( version: string ): string {
-	debug( `Resolving PHP version '${ version }'` );
+export function resolvePhpVersion( version: string | number ): string {
+	if ( typeof version !== 'string' ) {
+		version = version?.toString() || '';
+	}
+
+	debug( `Resolving PHP version %j`, version );
 
-	if ( typeof version === 'string' && version.startsWith( 'image:' ) ) {
+	if ( version.startsWith( 'image:' ) ) {
 		return version;
 	}
```

Comment:
We cannot drop these lines because of backward compatibility. However, I doubt that anyone used this undocumented functionality.